openbb_terminal/reports/reports_controller.py

```diff
@@ -209,27 +213,66 @@ def switch(self, an_input: str):
                 "stored",
                 report_output_name,
             )
+            # only a single backslash appears in the .py file otherwise
+            notebook_output = notebook_output.replace("\\", "\\\\")
+            notebook_output = notebook_output.replace("\\", "\\\\")
 
+            # gather params from user
             d_report_params = {}
-            for idx, args in enumerate(params):
-                d_report_params[args] = other_args[idx]
+            for idx in range(len(params) - 1):
+                d_report_params[params[idx]] = other_args[idx]
 
             d_report_params["report_name"] = notebook_output
 
-            result = pm.execute_notebook(
-                notebook_template + ".ipynb",
-                notebook_output + ".ipynb",
-                parameters=d_report_params,
-                kernel_name="python3",
+            notebook_file = os.path.join(
+                os.path.dirname(os.path.abspath(__file__)), report_to_run
             )
+            notebook_content = open(notebook_file + ".ipynb").read()
+
+            # replace params in notebook
+            if "parameters" in notebook_content:
+                params_to_change = self.d_params[report_to_run][1]
+                for idx, param in enumerate(params_to_change):
+                    notebook_content = notebook_content.replace(
+                        param, d_report_params[params[idx]], 1
+                    )
+
+                notebook_file_copy = notebook_file + "copy"
 
-            if not result["metadata"]["papermill"]["exception"]:
+                with open(notebook_file_copy + ".ipynb", "w") as file:
+                    file.write(notebook_content)
+
+            # use nbconvert to switch to .py
+            os.system(f"jupyter nbconvert --to script --no-prompt {notebook_file_copy}")
```

Comment:
can you use the nbconvert python api insteas of an os.system call?
see https://nbconvert.readthedocs.io/en/latest/nbconvert_library.html

Reply:
Thanks for this, I wasn't able to figure this out from their github

Reply:
you can also look into nbconverts tooling to load and parse the notebook and get the cells